src/libraries/System.Text.Json/Common/JsonSimpleNamingPolicy.cs

```diff
@@ -0,0 +1,147 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+using System.Buffers;
+using System.Globalization;
+
+namespace System.Text.Json
+{
+    internal sealed class JsonSimpleNamingPolicy : JsonNamingPolicy
+    {
+        private readonly bool _lowercase;
+        private readonly char _boundary;
+
+        internal JsonSimpleNamingPolicy(bool lowercase, char boundary) =>
+            (_lowercase, _boundary) = (lowercase, boundary);
```

Comment:
JIT Codegen suboptimal.
see [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgNXIYG8aGeGAHKAEsAbtgwwGsbABMIAOwA2ATwbAIEBQwD6CiAHcYUMHhgBubrwEixEqbMUqwAC2xRtagK5zprpeeq8DBY8gnLiUHLYmmwAFGoaDLoGRiZoDM6uqhBePlBKAJQMALwAfAwxOvqGxrgwaVqe3r6FReVJ1alZOc3+AL409EykrMNcAZZCouKSMDLyylkJlck1ZsH8kzYzcw7pLm4N2U15/oHroeGR0aRx6prtKbVpGW6NuQXrY4GByx21xYkqo81uNvu4ju8AW9fKdeP1qL0gA===).

Seems like #75206.

/cc @SingleAccretion

Reply:
This `ctor` is called only in for places and it happens ones for each, so I see no reason to change that.

Reply:
You're right about that. 
I mean it's more common to write that out though instead of doing a double assignment, so I just recommend my suggestion.

Performance-wise, this is not relevant in this specific case, but I mentioned single because it might be a known problem that is definitely more important in other places.

Reply:
I think we generally prefer two assignments in this code base. If we change conventions we should do it everywhere or follow them

Reply:
I would recommend following convention for constructors used in the rest of the codebase, regardless of any perf considerations.